core/src/main/scala/kafka/server/metadata/BrokerMetadataSnapshotter.scala

```diff
@@ -45,26 +45,37 @@ class BrokerMetadataSnapshotter(
    */
   private var _currentSnapshotOffset = -1L
 
+  /**
+   * The offset of the newest snapshot, or -1 if there hasn't been one. Accessed only under
+   * the object lock.
+   */
+  private var _latestSnapshotOffset = -1L
+
   /**
    * The event queue which runs this listener.
    */
   val eventQueue = new KafkaEventQueue(time, logContext, threadNamePrefix.getOrElse(""))
 
   override def maybeStartSnapshot(lastContainedLogTime: Long, image: MetadataImage): Boolean = synchronized {
-    if (_currentSnapshotOffset == -1L) {
+    if (_currentSnapshotOffset != -1) {
+      warn(s"Declining to create a new snapshot at ${image.highestOffsetAndEpoch()} because " +
```

Comment:
Do you think the log level is right for these messages? It seems like these are "normal" things that can happen and not an indication of a problem. Perhaps we can make them info or debug level.

Reply:
I agree with you here, info level is more suitable.